### PR TITLE
Keep Fit Results In Muon Analysis When New Data Is Loaded 

### LINKED ISSUE #31298: Muon results table has no memory
-	Load and fit a run (any run and function will do)
-	Step to the next run and fit that too, using the same function
-	Repeat for a few more runs
-	Go to Results Table. Only one run is there (the last one fitted).

### PR DESCRIPTION
**Pull request by Matt-Cumber**
**Description of work.**
Removed some unnecessary lines of code that were clearing all fits when not needed from the fitting context

**To test:**

1. Open Muon Analysis
2. Load a run (e.g. MUSR62260)
3. Go to the fitting tab and fit any function (e.g. Linear background) and press fit (doesn't need to be a good fit)
4. Go to the results tab and verify the fit workspace is present
5. Go back to Fitting tab
6. Step to the next run by clicking the arrow next to run number
7. Once the data has loaded click fit once again and do this for several runs
8. Now go to the results tab
9. The results tab should include the results from all the fits you performed not just the last one
10. On the fitting tab click undo fit
11. The latest fit should now be removed from the results table
12. On the home tab change instrument
13. All fits should be removed from the results tab

Fixes #31298 

*This does not require release notes* because **this is a bug from this release not present in the previous release**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by robertapplin — CHANGES_REQUESTED** [review]
Thanks for this fix, just a few comments that might need some changing. When you do a fit and then change an instrument, all of the fits should be removed, but this doesn't happen at the moment. I think this can be fixed quite easily here https://github.com/mantidproject/mantid/blob/c4a8230fec10d95155c302fd3cf25c28e8012183/scripts/Muon/GUI/Common/fitting_widgets/general_fitting/general_fitting_presenter.py#L51

**Review by robertapplin — APPROVED** [review]
Good stuff, thanks for fixing this

File: Testing/SystemTests/tests/qt/PythonInterfacesStartupTest.py [hunk 1]
```diff
@@ -64,8 +66,10 @@ def _attempt_to_open_python_interface(self, interface_script):
         set_application_name(interface_script)
         # Prevents a QDialog popping up when opening certain interfaces
         set_instrument(interface_script)
+        # Some interfaces need to be closed after opening. This also ensures they process all their events
+        close_code = CLOSE_CODE_SWITCHER.get(interface_script, "")
 
         try:
-            exec(open(os.path.join(self._interface_directory, interface_script)).read())
```
Comment (DavidFair) [inline Testing/SystemTests/tests/qt/PythonInterfacesStartupTest.py:69]: This has a real code smell to it, as were effectively echoing concatenated code into an exec. Is there no way we can call `.close()` in some sort of Qt hook when the window is closing, or am I misunderstanding what it's doing?

**Review by DavidFair — APPROVED** [review]
On discussion it would be best for Rob to jointly tackle this: However, as he is on leave currently, this PR fixes a problem in the release and it's only a code smell rather than a problem lets :shipit:

**Pull request merged**